src/mlpack/tests/xgboost_test.cpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * @file tests/xgboost_test.cpp
+ * @author Rishabh Garg
+ *
+ * Tests for the XGBoost class and related classes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/xgboost/loss_functions/sse_loss.hpp>
+
+#include "catch.hpp"
+#include "serialization.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ensemble;
+
+/**
+ * Test that the initial prediction is calculated correctly for SSE loss.
+ */
+TEST_CASE("SSEInitialPredictionTest", "[XGBTest]")
+{
+  arma::vec values = {1, 3, 2, 2, 5, 6, 9, 11, 8, 8};
+
+  double initPred = 5.5;
+
+  SSELoss Loss;
+  REQUIRE(Loss.InitialPrediction(values) == initPred);
+}
```

Comment:
It wouldn't hurt to add some sanity checks to ensure that, e.g., empty input doesn't crash. :)

Reply:
Yeah! Will add the checks 👍